python/src/numerous/user.py

```diff
@@ -0,0 +1,42 @@
+"""Module for handling user-related functionality."""
+
+from typing import Any, Callable, Dict
+from numerous.collection import NumerousCollection, collection
+import base64
+import json
+
+class User:
+    def __init__(self, cookie_getter: Callable[[], Dict[str, Any]]) -> None:
+        self._cookie_getter = cookie_getter
+
+    def _get_user_info(self) -> Dict[str, Any]:
+        """Decode and return the user info from the cookie."""
+        cookies = self._cookie_getter()
+        user_info_b64 = cookies.get("numerous_user_info")
+        if not user_info_b64:
+            return {}
+        try:
+            user_info_json = base64.b64decode(user_info_b64).decode('utf-8')
+            return json.loads(user_info_json)
+        except (ValueError, json.JSONDecodeError):
+            return {}
+
+    @property
+    def id(self) -> str:
+        """Get the user ID from the decoded cookie or return a default value."""
+        user_info = self._get_user_info()
+        return user_info.get("user_id") or "local-user"
+    
+    @property
+    def name(self) -> str:
+        """Get the user name from the decoded cookie or return a default value."""
+        user_info = self._get_user_info()
+        return user_info.get("user_full_name") or "Local User"
+
+    def collection(self, collection_key: str) -> NumerousCollection|None:
+        """Get a user-specific collection."""
+        user_collection = collection("users").collection(self.id)
+        if user_collection is None:
+            error_message = "User collection not found"
+            raise ValueError(error_message)
+        return user_collection.collection(collection_key)
```

Comment:
I think I would move most of the logic into the session class, since the user information does not change. Then we do not need to have conditional logic every time we access the `name` or `id`.

I think the logic of providing fallback values for user ID and user name should be outside the user.

I also think we should just return the user collection itself, maybe? That seems more flexible.

```python
@dataclass
class User:
    id: str
    name: str

    @property
    def collection(self) -> NumerousCollection | None:
        user_collection collection("users").collection(self.id)

    @staticmethod
    def from_user_info(user_info: dict[str, Any]) -> "User":
        return User(id=user_info["user_id"], name=user_info["name"])
```

Reply:
This has been implemented close to the suggestion